urld025f21ac3_mvattipulusu_Innovation2013_tgz-pJ8-com_usps_data_ManifestFileTestJ8/src/com/evs/data/util/DataTypes.java

```diff
@@ -0,0 +1,1326 @@
+ * 
+ *    Rev 1.62   Jun 12 2009 16:37:58   sutantod
+ * Addd new method
+ * 
+ *    Rev 1.61   Jun 11 2009 16:00:46   huangk
+ * Modified isFileVersionNumber to accept 1.4 only.
+ * 
+ *    Rev 1.60   Jun 10 2009 13:45:56   gandhia
+ * STC logic updated for Samples and Manifest processing.
+ * 
+ *    Rev 1.59   Jun 09 2009 15:06:32   gandhia
+ * Updated STC for Express Mail
+ * 
+ *    Rev 1.58   May 28 2009 11:30:16   sutantod
+ * Validation on Express Mail RR followed by IM
+  * 
+ */
+
+package com.evs.data.util;
+
+import java.io.File;
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+
+public class DataTypes {
+
+    public static final int MANIFEST = 1;
+    public static final int STATS = 2;
+    public static final int PDA = 3;
+    public static final int EXTRACT = 4;
+    public static final int CEW = 5;
+    public static final int CEW_RECEIPT = 6;
+    public static final int CEW_MAILING = 7;
+    public static final int PRS_PDA = 8;
+    public static final int PASS = 9;  
+    public static final int CAD = 10;
+
+    public static void main(String[] args) {
+    	String code = "9AB";
+		System.out.println("Code-->[" + code + "], Return Value = [" + padFedAgencyCostCode("9AB", false, 8) + "]");
+		if (isEmptyStr(" L ")) System.out.println("Empty");
+	}
+	public static String massageProcessingCategory(String processingCategory) {
+
+		if ("".equals(processingCategory)) {
+			processingCategory = Constants.PROCESSING_CATEGORY_EVS_PARCEL_FIRSTCLASS;
+		}
+		return processingCategory;
+	}
+    public static boolean isEmptyStr(String s) {
+    	return (s == null || s.trim().equals("") ? true : false); 
+    }
+    /**
+     * Return a date/time format for a specific file format.
+     */
+    public static SimpleDateFormat getDateTimeConvert(int fileType)
+    {
+        if(fileType == MANIFEST)
+        {
+            return new SimpleDateFormat("yyyyMMddHHmmss");
+        }
+        else if(fileType == STATS)
+        {
+            return new SimpleDateFormat("yyyyMMdd");
+        }
+        else if(fileType == PDA)
+        {
+            return new SimpleDateFormat("MMddyyyy");
+        }
+        else if(fileType == PRS_PDA)
+        {
+			return new SimpleDateFormat("MMddyyyy");
+        }
+        else if(fileType == EXTRACT)
+        {
+            return new SimpleDateFormat("yyyyMMddHHmm");
+        }
+        else if(fileType == CEW_RECEIPT)
+        {
+            return new SimpleDateFormat("yyyyMMddHHmmss");
+        }
+        else if(fileType == CEW_MAILING)
+        {
+            return new SimpleDateFormat("yyyyMMdd");
+        }
+        else if(fileType == PASS)
+        {
+        	return new SimpleDateFormat("yyyyMMdd");
+        }
+        else if(fileType == CAD)
+        {
+        	return new SimpleDateFormat("MMddyyyy");
+        }
+        else
+        {
+            return null;
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 125:148, 195:218.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.